acceptance/lib/puppet/acceptance/environment_utils.rb

```diff
@@ -0,0 +1,44 @@
+module Puppet
+  module Acceptance
+    module EnvironmentUtils
+
+     # generate a random string of 6 letters and numbers.  NOT secure
+      def random_string
+        [*('a'..'z'),*('0'..'9')].shuffle[0,8].join
+      end
+      private :random_string
+
+      # if the first test to call this has changed the environmentpath, this will cause trouble
+      #   maybe not the best idea to memoize this?
```

Comment:
i think there is a tradeoff here between speed/efficiency and state.  every call to host.puppet['blah'] runs puppet on the host with --configprint.  this is slow.

i think this is a good option, when the vast majority just wants the default environmentpath